### PR TITLE
Fix bug 1635743 [API] Exclude system projects from default

### PR DESCRIPTION
**Pull request by vishnutk**
Introduces includeSystem parameter which is by default false. Tested with below graphql query on local box and getting expected result.

```
query {
  projects(includeDisabled: true, includeSystem: false) {
    name
  }
  locale(code:"mr") {
    name
    googleTranslateCode
    
    localizations(includeSystem: true) {
      project {
        name
      }
    }
  }
}
```
@mathjazz r?

### CODE REVIEW

File: pontoon/api/schema.py [hunk 1]
```diff
@@ -100,29 +100,40 @@ class Meta:
...
-        if include_disabled:
+        if include_disabled and include_system:
             return qs.all()
+        elif include_disabled and not include_system:
+            return qs.filter(project__system_project=False)
+        elif not include_disabled and include_system:
+            return qs.filter(project__disabled=False)
 
-        return qs.filter(project__disabled=False)
+        return qs.filter(project__disabled=False, project__system_project=False)
```
Comment (mathjazz) [inline pontoon/api/schema.py:118]: I think you can simplify this entire block to:
```python
return qs.filter(
    project__disabled=include_disabled,
    project__system_project=include_system,
)
```

File: pontoon/api/schema.py [hunk 2]
```diff
@@ -132,10 +143,14 @@ def resolve_projects(obj, info, include_disabled):
...
-        if include_disabled:
+        if include_disabled and include_system:
             return qs.all()
+        elif include_disabled and not include_system:
+            return qs.filter(system_project=False)
+        elif not include_disabled and include_system:
+            return qs.filter(disabled=False)
 
-        return qs.filter(disabled=False)
+        return qs.filter(disabled=False, system_project=False)
```
Comment (mathjazz) [inline pontoon/api/schema.py:153]: I think you can simplify this entire block to:
```python
return qs.filter(disabled=include_disabled, system_project=include_system)
```

File: pontoon/api/schema.py (hunk 1 above)
Comment (vishnutk) [reply pontoon/api/schema.py:118]: This is not giving correct result. Reason is QuerySet filter does AND on the condition. As per my understanding we need below result, correct?

// disabled = true and system = true - should return all the project
// disabled = true and system = false - should return all the project except system
// disabled = false and system = true - should return all the project except disabled
// disabled = false and system = false - should return all the project except disabled and system

I am not able to write complex query which will give this result with .filter/.exlude. Thus its better to keep this block.

Comment (mathjazz) [reply pontoon/api/schema.py:122]: That's a great point!

I'm afraid the current solution doesn't scale very well. Imagine we'll add one or more conditions we'd like to exclude by default.

How about something like this?

```python
qs = obj.project_locale

records = qs.filter(disabled=False, system_project=False)

if include_disabled:
    records = records | qs.filter(project__disabled=True)
if include_system:
    records = records | qs.filter(project__system_project=True)

return records.distinct()
```

**Pull request merged**